protocols/gossipsub/src/behaviour.rs

```diff
@@ -3140,16 +3139,16 @@ where
         _: Endpoint,
         _: PortUse,
     ) -> Result<THandler<Self>, ConnectionDenied> {
-        let connected_peer = self
-            .connected_peers
-            .entry(peer_id)
-            .or_insert(PeerConnections {
-                kind: PeerKind::Floodsub,
-                connections: vec![],
-                sender: Sender::new(self.config.connection_handler_queue_len()),
-                topics: Default::default(),
-                dont_send: LinkedHashMap::new(),
-            });
+        let connected_peer = self.connected_peers.entry(peer_id).or_insert(PeerDetails {
+            kind: PeerKind::Floodsub,
+            connections: vec![],
+            // Diverging from the go implementation we only want to consider a peer as outbound peer
+            // if its first connection is outbound.
+            outbound: !self.px_peers.contains(&peer_id),
```

Comment:
We used to also check that `other_established == 0`. Is that not needed anymore?

Reply:
in this case it's redundant I think, We are introducing a new `PeerDetails`  which means it's the first time the peer is connecting right?

Reply:
Ah yes if course, I didn't think about that.

Reply:
no worries Elena, thanks for your check!